homeassistant/components/lock/kiwi.py

```diff
@@ -34,8 +34,12 @@
 
 def setup_platform(hass, config, add_devices, discovery_info=None):
     """Set up the KIWI lock platform."""
-    from kiwiki import KiwiClient
-    kiwi = KiwiClient(config[CONF_USERNAME], config[CONF_PASSWORD])
+    from kiwiki import KiwiClient, KiwiException
+    try:
+        kiwi = KiwiClient(config[CONF_USERNAME], config[CONF_PASSWORD])
+    except KiwiException as e:
+        _LOGGER.error(e.msg)
+        return False
```

Comment:
That is horrible to read. I prefer `return False` - there is no room for misinterpretation.

Reply:
Nothing is checking the return value.

Reply:
Check the platform example - 
`False` is returned here on _login failed_. It makes sense even if its not checked because  it improves the readability. httpsevelopers.home-assistant.io/docs/en/creating_platform_example_light.html

Reply:
That example needs updating.

It doesn't improve readability. It gives the wrong impression what happens when the function returns.

Reply:
Ok, I created an Issue (https://github.com/home-assistant/developers.home-assistant/issues/21) and will change it to `return`